cockatrice/src/dlg_connect.cpp

```diff
@@ -132,24 +132,85 @@ DlgConnect::DlgConnect(QWidget *parent)
         previousHostButton->setChecked(true);
     else
         newHostButton->setChecked(true);
+    
+    connect(previousHosts, SIGNAL(currentIndexChanged(const QString &)), this, SLOT(updateDisplayInfo(const QString &)));
 
     playernameEdit->setFocus();
 }
 
+void DlgConnect::actSaveConfig()
+{
+    if (!settingsCache->servers().updateExistingServer(saveEdit->text(), hostEdit->text(), portEdit->text(), playernameEdit->text(), passwordEdit->text(), savePasswordCheckBox->isChecked()))
+        settingsCache->servers().addNewServer(saveEdit->text(), hostEdit->text(), portEdit->text(), playernameEdit->text(), passwordEdit->text(), savePasswordCheckBox->isChecked());
+
+    rebuildComboBoxList();
+}
+
+void DlgConnect::rebuildComboBoxList()
+{
+    previousHosts->clear();
+
+    UserConnection_Information uci;
+    savedHostList = uci.getServerInfo();
+
+    if (savedHostList.size() == 1)
+    {
+        settingsCache->servers().addNewServer("Woogerworks Cockatrice", "cockatrice.woogerworks.com", "4747", "", "", false);
+        settingsCache->servers().addNewServer("Chickatrice Cockatrice", "chickatrice.net", "4747", "", "", false);
+        settingsCache->servers().addNewServer("Cockatric.es Cockatrice", "cockatric.es", "4747", "", "", false);
+        settingsCache->servers().addNewServer("Tetrarch Cockatrice", "mtg.tetrarch.co", "4747", "", "", false);
+    }
+    savedHostList = uci.getServerInfo();
+
+    int i = 0;
+    foreach (UserConnection_Information tmp, savedHostList)
+    {
+        QString saveName = tmp.getSaveName();
+        if (saveName.size())
+        {
+            previousHosts->addItem(saveName);
+            
+            if (settingsCache->servers().getPrevioushostName() == saveName)
+                previousHosts->setCurrentIndex(i);
+            
+            i++;
+        }
+    }
+}
+
 
 void DlgConnect::previousHostSelected(bool state) {
     if (state) {
-        hostLabel->setDisabled(true);
-        hostEdit->setDisabled(true);
         previousHosts->setDisabled(false);
     }
 }
 
+void DlgConnect::updateDisplayInfo(const QString &saveName)
+{
+    UserConnection_Information uci;
+    QStringList data = uci.getServerInfo(saveName);
+
+    if (saveEdit == NULL)
```

Comment:
Prefer `nullptr` over `NULL` in the future. This is fine for now though, not worth triggering CI again for it